vocabularies/Analytics.xml

```diff
@@ -176,6 +179,114 @@
         </Annotation>
       </Annotations>
 
+      <Function Name="AutoExpand" IsBound="true" EntitySetPath="InputSet">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.IsSignature" />
+        <Annotation Term="Core.Description" String="`$apply` transformation that expands an unnamed leveled hierarchy with custom aggregation of certain properties" />
+        <Parameter Name="InputSet" Type="Collection(Edm.EntityType)" Nullable="false">
+          <Annotation Term="Core.Description" String="Entity set to be processed" />
+        </Parameter>
+        <Parameter Name="Levels" Type="Collection(Analytics.AutoExpandLevel)" Nullable="false">
+          <Annotation Term="Core.Description" String="Collection of aggregation levels forming the visible part of a leveled hierarchy" />
+          <Annotation Term="Core.LongDescription">
+            <String>Each element in the collection defines the property names for one level.
+            A property must not referenced by more than one level.
+            The first element in the collection defines the property names of the coarsest level,
+            the following elements define the property names of consecutively finer-grained aggregation levels.
+            The function result comprises the leveled hierarchy with these levels in preorder.
+            All referenced properties must be groupable.</String>
+          </Annotation>
+        </Parameter>
+        <Parameter Name="LeafLevel" Type="Collection(Edm.String)" Nullable="false">
+          <Annotation Term="Core.Description" String="A possibly empty set of property names that constitute, together with the property names from `Levels`, the leaf level of the leveled hierarchy" />
+          <Annotation Term="Core.LongDescription">
+            <String>Property names that occur in `Levels` must not be repeated here.
+            All referenced properties must be groupable.</String>
+          </Annotation>
+        </Parameter>
+        <Parameter Name="Aggregation" Type="Collection(Edm.String)" Nullable="false">
+          <Annotation Term="Core.Description" String="Properties to aggregate for all result entries on all levels" />
+          <Annotation Term="Core.LongDescription">
+            <String>All properties in this collection must be custom aggregates.</String>
+          </Annotation>
+        </Parameter>
+        <Parameter Name="SiblingOrder" Type="Collection(Analytics.AutoExpandSiblingOrder)" Nullable="false">
+          <Annotation Term="Core.Description" String="Sort specification to apply to all direct descendants of a given entry in the resulting leveled hierarchy" />
+        </Parameter>
+        <Parameter Name="BeforeAggregationFilter" Type="Edm.String" Nullable="false">
+          <Annotation Term="Core.Description" String="Expression valid for `filter` transformation to restrict the input set before any further procressing" />
```

Comment:
Can't you achieve the same thing with a filter() transformation beforehand, in the same $apply? For TopLevels, we do likewise with orderby() or search() or filter().

Reply:
Yes, but then the backend would have to take all these transformations together and process them as one "auto-expand" transformation. (At least with the Analytical Engine, the filter is part of the auto-expand operation.)

If the backend agrees, we can change this.

Reply:
If the backend disagrees, do we need support for search as well? Or is that too fuzzy for analytics?